From 58a3a7ba49784f1baa7b4f7d6680211897a8b3f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Jutteau?= <jerome.jutteau@outscale.com>
Date: Tue, 5 Oct 2021 10:29:26 +0200
Subject: [PATCH] patch: remove omitempty json tag of
 BsuToCreate.DeleteOnVmDeletion
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Probably due to an old genetator issue[1], value for booleans are omited when value is set to false.
This also occurs when `default: true` is added to openapi yaml description.
This quick workaround removes the omitempty tag to force its value.

[1] https://github.com/swagger-api/swagger-codegen/issues/7391

Signed-off-by: Jérôme Jutteau <jerome.jutteau@outscale.com>
---
 osc/model_bsu_to_create.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/osc/model_bsu_to_create.go b/osc/model_bsu_to_create.go
index 523d679..5ea1bcc 100644
--- a/osc/model_bsu_to_create.go
+++ b/osc/model_bsu_to_create.go
@@ -13,7 +13,7 @@ package osc
 // BsuToCreate Information about the BSU volume to create.
 type BsuToCreate struct {
 	// By default or if set to true, the volume is deleted when terminating the VM. If false, the volume is not deleted when terminating the VM.
-	DeleteOnVmDeletion bool `json:"DeleteOnVmDeletion,omitempty"`
+	DeleteOnVmDeletion bool `json:"DeleteOnVmDeletion"`
 	// The number of I/O operations per second (IOPS). This parameter must be specified only if you create an `io1` volume. The maximum number of IOPS allowed for `io1` volumes is `13000`.
 	Iops int32 `json:"Iops,omitempty"`
 	// The ID of the snapshot used to create the volume.
-- 
2.33.0

